peer_notify.go

```diff
@@ -0,0 +1,91 @@
+package pubsub
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p/core/event"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+func (ps *PubSub) watchForNewPeers(ctx context.Context) {
+	// We don't bother subscribing to "connectivity" events because we always run identify after
+	// every new connection.
```

Comment:
I _think_ this is right but I'm not 100% certain. I remember remember running into some issues with this with respect to bitswap...

cc @sukunrt do you think the code here is sufficient?

Reply:
Yeah this looks fine. We run identify after connecting on every connection.